File: src/main/java/com/pk/YourSoccerField/controller/UserController.java
package com.pk.YourSoccerField.controller;

import com.pk.YourSoccerField.service.UserService;
import com.pk.YourSoccerField.service.dtoModel.UserDTO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;

@RestController
@RequestMapping("/api")
public class UserController {

    private final UserService userService;

    @Autowired
    public UserController(UserService userService) {
        this.userService = userService;
    }

    @PostMapping(
            value = "/register",
            consumes = MediaType.APPLICATION_JSON_VALUE,
            produces = MediaType.APPLICATION_JSON_VALUE)
    public ResponseEntity<?> register(@RequestBody @Valid UserDTO userDTO) {
        return new ResponseEntity<>(
                userService.createUser(userDTO),
                HttpStatus.CREATED
        );
    }

}

File: api/src/main/kotlin/com/pk/ysf/api/controller/UserController.kt
package com.pk.ysf.api.controller

import com.pk.ysf.api.model.dto.LoginResponse
import com.pk.ysf.api.model.dto.UserInput
import com.pk.ysf.api.model.dto.UserReference
import com.pk.ysf.api.service.spec.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.net.URI
import javax.servlet.http.HttpServletRequest

@RestController
@RequestMapping("\${spring.data.rest.base-path}/users")
class UserController @Autowired constructor(
        private val userService: UserService
) {

    @GetMapping
    fun login(
            @RequestParam(defaultValue = "") username: String,
            @RequestParam(defaultValue = "") password: String
    ): ResponseEntity<LoginResponse> =
            ResponseEntity.ok(userService.login(username, password))

    @PostMapping
    fun register(
            @RequestBody userInput: UserInput,
            request: HttpServletRequest
    ): ResponseEntity<UserReference> {
        val createdUser: UserReference = userService.create(userInput)

        return ResponseEntity
                .created(URI.create(request.requestURI + "/${createdUser.code}"))
                .body(userService.create(userInput))
    }

}